tensorly/tenalg/tenalg_utils.py

```diff
@@ -0,0 +1,58 @@
+def _validate_contraction_modes(shape1, shape2, modes, batched_modes=False):
+    """Takes in the contraction modes (for a tensordot) and validates them
+    
+    Parameters
+    ----------
+    modes : int or tuple[int] or (modes1, modes2)
```

Comment:
How do you distinguish between the second and third case? 
```python3
a_shape  = (5, 4, 5)
b_shape  = (8, 4, 5)
_validate_contraction_modes(a_shape, b_shape, (1,2))
```
currently returns an error. I think we should remove the second case from the docstring. This will also keep it consistent with `tl.tensordot`.

Reply:
I was hesitant about this very usecase. It is convenient to be able to pass a tuple in. But then the case where it's of length 2 is indeed ambiguous, so I left the error. Might indeed be best to keep it consistent with `tensordot`.